test/integration/integration_test.cc

```diff
@@ -19,19 +19,20 @@ INSTANTIATE_TEST_CASE_P(IpVersions, IntegrationTest,
 
 TEST_P(IntegrationTest, RouterNotFound) { testRouterNotFound(); }
 
-TEST_P(IntegrationTest, RouterNotFoundBodyNoBuffer) {
-  testRouterNotFoundWithBody(lookupPort("http"));
-}
+TEST_P(IntegrationTest, RouterNotFoundBodyNoBuffer) { testRouterNotFoundWithBody(); }
 
 TEST_P(IntegrationTest, RouterNotFoundBodyBuffer) {
-  testRouterNotFoundWithBody(lookupPort("http_buffer"));
+  config_helper_.addFilter(ConfigHelper::DEFAULT_BUFFER_FILTER);
+  testRouterNotFoundWithBody();
 }
 
 TEST_P(IntegrationTest, RouterRedirect) { testRouterRedirect(); }
 
 TEST_P(IntegrationTest, DrainClose) { testDrainClose(); }
 
 TEST_P(IntegrationTest, ConnectionClose) {
+  config_helper_.addFilter(ConfigHelper::DEFAULT_HEALTH_CHECK_FILTER);
+  initialize();
   executeActions({[&]() -> void { codec_client_ = makeHttpConnection(lookupPort("http")); },
```

Comment:
could initialize() somehow be merged into executeActions so we don't have to call it in every test as well?

Reply:
Why do we still have `executeActions`? I thought this was vestigial of an earlier testing regime and we can just write straight line test code now.

Reply:
I don't remember why we have it. It might not be needed any more. We can get rid of it if it doesn't do anything. I guess my main point is it would be nice to not have to call `initialize()` in every test, but not a big deal.

Reply:
Ooohh.  I like that idea but we can't do it easily today because SSL tests create a client connection before they executeAction, and that requires Envoy to be started.

First push replaces all of the low hanging fruit, second does SSL connection creation via custom functions so we can do lazy eval and late initialize() there as well.